src/main/java/com/redhat/devtools/lsp4ij/features/rename/LSPRenameRefactoringDialog.java

```diff
@@ -192,4 +228,90 @@ protected void dispose() {
         super.dispose();
     }
 
+    @NotNull
+    private static Set<PsiReference> getExternalReferences(@NotNull PsiFile file, int offset) {
+        Set<PsiReference> externalReferences = new LinkedHashSet<>();
+
+        Document document = LSPIJUtils.getDocument(file);
+        TextRange wordTextRange = document != null ? LSPIJUtils.getWordRangeAt(document, file, offset) : null;
+        if (wordTextRange != null) {
+            LSPPsiElement wordElement = new LSPPsiElement(file, wordTextRange);
+            String wordText = wordElement.getText();
+            if (StringUtil.isNotEmpty(wordText)) {
+                // When testing, just collect references on the current thread
+                if (ApplicationManager.getApplication().isUnitTestMode()) {

```

Comment:
I don't love this, but it's important that the textual search occur under a modal progress indicator, and that doesn't work well with the rename refactoring unit tests. The same logic occurs either way, but in the real app it occurs under a modal progress indicator, and during tests it occurs on the current thread.